tests/test_multipart.py

```diff
@@ -599,6 +599,18 @@ async def test_read_form(self, newline: Any) -> None:
             result = await obj.form()
         assert [("foo", "bar"), ("foo", "baz"), ("boo", "")] == result
 
+    async def test_read_form_invalid_utf8(self, newline: Any) -> None:
+        data = b"\xff%s--:--" % newline
+        with pytest.raises(ValueError):
```

Comment:
Same here: plz add a `match=` arg.

Reply:
I haven't added a `match=` here because I now catch two exceptions: the one raised by the C parser and the one raised by the Python parser, which have different texts.